api_db.php

```diff
@@ -378,35 +378,31 @@ function SQLite3_connect($trytoreconnect)
 			$interval = $q;
 	}
 
+	// Round $from and $until to match the requested $interval
+	$from = intval((intval($_GET['from'])/$interval)*$interval);
+	$until = intval((intval($_GET['until'])/$interval)*$interval);
+
 	// Count permitted queries in intervals
 	$stmt = $db->prepare('SELECT (timestamp/:interval)*:interval interval, COUNT(*) FROM queries WHERE (status != 0 )'.$limit.' GROUP by interval ORDER by interval');
-	$stmt->bindValue(":from", intval($_GET['from']), SQLITE3_INTEGER);
-	$stmt->bindValue(":until", intval($_GET['until']), SQLITE3_INTEGER);
+	$stmt->bindValue(":from", $from, SQLITE3_INTEGER);
+	$stmt->bindValue(":until", $until, SQLITE3_INTEGER);
 	$stmt->bindValue(":interval", $interval, SQLITE3_INTEGER);
 	$results = $stmt->execute();
 
-	// Parse the DB result into graph data, filling in missing sections with zero
-	function parseDBData($results, $interval) {
+	// Parse the DB result into graph data, filling in missing interval sections with zero
+	function parseDBData($results, $interval, $from, $until) {
 		$data = array();
-		$min = null;
-		$max = null;
 
 		if(!is_bool($results)) {
 			// Read in the data
 			while($row = $results->fetchArray()) {
-				// Get min and max timestamps
-				if($min === null || $min > $row[0])
-					$min = $row[0];
-
-				if($max === null || $max < $row[0])
-					$max = $row[0];
-
-				// Get the non-zero graph data
+				// $data[timestamp] = value_in_this_interval
 				$data[$row[0]] = intval($row[1]);
 			}
 
 			// Fill the missing intervals with zero
-			for($i = $min; $i < $max; $i += $interval) {
+			// Advance in steps of interval
+			for($i = $from; $i < $until; $i += $interval) {
```

Comment:
This function is now changed to not only fill in missing intervals between the first and last timestamp of the data *returned by the database* but to fill in missing intervals between the first and last timestamp of the data *requested by the user* (always `>=` than what the database returns).